cache/memcached_client.go

```diff
@@ -291,15 +307,16 @@ func newMemcachedClient(
 		func() float64 { return 1 },
 	)
 
-	// As soon as the client is created it must ensure that memcached server
-	// addresses are resolved, so we're going to trigger an initial addresses
-	// resolution here.
-	if err := c.resolveAddrs(); err != nil {
-		return nil, err
-	}
+	return c, nil
+}
+
+// Initialize will start the background DNS resolution periodically and do an initial DNS resolution.
```

Comment:
Can you note here that the background goroutine is started even if the initial resolution fails?